types/date-arithmetic/index.d.ts

```diff
@@ -1,38 +1,77 @@
+export function month(date: Date): number;
+export function month(date: Date, value: number): Date;
+export function year(date: Date): number;
+export function year(date: Date, value: number): Date;
+export function decade(date: Date): number;
+export function decade(date: Date, value: number): Date;
+export function century(date: Date): number;
+export function century(date: Date, value: number): Date;
 
-    /** Subtract specified amount of units from a provided date and return new date as a result */
-    subtract(date: Date, num: number, unit: Unit): Date;
+/** Add specified amount of units to a provided date and return new date as a result */
+export function add(date: Date, num: number, unit: Unit): Date;
 
-    /** Compare two dates and return true if they are equal */
-    eq(date: Date, date2: Date): Boolean;
+/**
+ * The opposite of `startOf`
+ */
+export function endOf(date: Date, unit: Unit): Date;
+/**
+ * Return a new date with the relevent date parts zero'd out.
+ * You only need to provide a firstOfWeek when the unit is 'week'
+ */
+export function startOf(date: Date, unit: 'week', firtOfWeek: StartOfWeeek): Date;
+export function startOf(date: Date, unit: Exclude<Unit, 'week'>): Date;
 
-    /** Compare two dates and return false if they are equal */
-    neq(date: Date, date2: Date): Boolean;
+/** Subtract specified amount of units from a provided date and return new date as a result */
+export function subtract(date: Date, num: number, unit: Unit): Date;
 
-    /** Compare two dates and return true if date is greater than date2 */
-    gt(date: Date, date2: Date): Boolean;
+/** Compare two dates and return true if they are equal */
+export function eq(date: Date, date2: Date): boolean;
 
-    /** Compare two dates and return true if date is greater or equal than date2 */
-    gte(date: Date, date2: Date): Boolean;
+/** Compare two dates and return false if they are equal */
+export function neq(date: Date, date2: Date): boolean;
 
-    /** Compare two dates and return true if date is less than date2 */
-    lt(date: Date, date2: Date): Boolean;
+/** Compare two dates and return true if date is greater than date2 */
+export function gt(date: Date, date2: Date): boolean;
 
-    /** Compare two dates and return true if date is less or equal than date2 */
-    lte(date: Date, date2: Date): Boolean;
-}
+/** Compare two dates and return true if date is greater or equal than date2 */
+export function gte(date: Date, date2: Date): boolean;
 
-declare module 'dateArithmetic' {
-    const dateArithmetic: dateArithmeticStatic;
-    export = dateArithmetic;
-}
+/** Compare two dates and return true if date is less than date2 */
+export function lt(date: Date, date2: Date): boolean;
+
+/** Compare two dates and return true if date is less or equal than date2 */
+export function lte(date: Date, date2: Date): boolean;
+
+export type StartOfWeeek = 0 | 1 | 2 | 3 | 4 | 5 | 6;
```

Comment:
```suggestion
export type StartOfWeek = 0 | 1 | 2 | 3 | 4 | 5 | 6;
```

Reply:
Ouch, fixed, amended in other places,
thx!